tests/test_client.py

```diff
@@ -578,3 +595,357 @@ def test_history_from_json():
     assert json_format.MessageToDict(history) == json_format.MessageToDict(
         _history_from_json(json_format.MessageToDict(history_from_dict))
     )
+
+
+async def test_schedule_basics(
+    client: Client, worker: ExternalWorker, env: WorkflowEnvironment
+):
+    if env.supports_time_skipping:
+        pytest.skip("Java test server doesn't support schedules")
+    await assert_no_schedules(client)
+
+    # Create a schedule with a lot of stuff
+    schedule = Schedule(
+        action=ScheduleActionStartWorkflow(
+            "kitchen_sink",
+            KSWorkflowParams(actions=[KSAction(result=KSResultAction("some result"))]),
+            id=f"workflow-{uuid.uuid4()}",
+            task_queue=worker.task_queue,
+            execution_timeout=timedelta(hours=1),
+            run_timeout=timedelta(hours=2),
+            task_timeout=timedelta(hours=3),
+            retry_policy=RetryPolicy(maximum_attempts=20),
+            memo={"memokey1": "memoval1"},
+        ),
+        spec=ScheduleSpec(
+            calendars=[
+                ScheduleCalendarSpec(
+                    # Intentionally set step 1 though 0 and 1 are the same to prove
+                    # that step comes back as sent not as defaulted (other 0 values
+                    # for step don't come back as 1)
+                    second=(ScheduleRange(1, step=1),),
+                    minute=(ScheduleRange(2, 3),),
+                    hour=(ScheduleRange(4, 5, 6),),
+                    day_of_month=(ScheduleRange(7),),
+                    month=(ScheduleRange(9),),
+                    year=(ScheduleRange(2080),),
+                    # Intentionally leave day of week absent to check default
+                    # day_of_week=[ScheduleRange(1)],
+                    comment="spec comment 1",
+                )
+            ],
+            intervals=[
+                ScheduleIntervalSpec(
+                    every=timedelta(days=10),
+                    offset=timedelta(days=2),
+                )
+            ],
+            cron_expressions=["0 12 * * MON"],
+            skip=[ScheduleCalendarSpec(year=(ScheduleRange(2050),))],
+            start_at=datetime(2060, 7, 8, 9, 10, 11, tzinfo=timezone.utc),
+            jitter=timedelta(seconds=80),
+        ),
+        policy=SchedulePolicy(
+            overlap=ScheduleOverlapPolicy.BUFFER_ONE,
+            catchup_window=timedelta(minutes=5),
+            pause_on_failure=True,
+        ),
+        state=ScheduleState(note="sched note 1", paused=True, remaining_actions=30),
+    )
+    handle = await client.create_schedule(
+        f"schedule-{uuid.uuid4()}",
+        schedule,
+        memo={"memokey2": "memoval2"},
+    )
+
+    # Alter the schedule to be the expected from server
+    assert isinstance(schedule.action, ScheduleActionStartWorkflow)
+    # Args are encoded
+    schedule.action.args = await DataConverter.default.encode(schedule.action.args)
```

Comment:
did you mean to `assert` here >

Reply:
To, this is meant to set the "expected" schedule to what's on the server so the `assert desc.schedule == schedule` below does a full nested equality assertion of everything including this